plenum/common/stack_manager.py

```diff
@@ -163,7 +163,11 @@ def stackKeysChanged(self, txn, remoteName, nodeOrClientObj):
         # Removing remote so that the nodestack will attempt to connect
         rid = self.removeRemote(nodeOrClientObj.nodestack, remoteName)
 
-        verkey = txn[VERKEY]
+        if txn[VERKEY][0] == '~':  # abbreviated
+            verkey = cryptonymToHex(txn[TARGET_NYM]) + cryptonymToHex(txn[VERKEY][1:])
```

Comment:
This is incorrect, `cryptonymToHex(txn[TARGET_NYM]) + cryptonymToHex(txn[VERKEY][1:])` is different from `cryptonymToHex(txn[TARGET_NYM]+txn[VERKEY][1:])`. Confirm with this,
```
from plenum.common.signer_did import DidSigner
from plenum.common.util import cryptonymToHex

c = 10

for i in range(c):
	print(i)
	d = DidSigner()
	a = cryptonymToHex(d.identifier+d.verkey[1:])
	b = cryptonymToHex(d.identifier) + cryptonymToHex(d.verkey[1:])
	assert a == b, (a,b)
```